ITensorGaussianMPS/src/gmps.jl

```diff
@@ -303,6 +264,14 @@ end
 # Correlation matrix diagonalization
 #
 
+struct Pairing
```

Comment:
What about the name `BCS`? I've also seen the name "Generalized Hartree-Fock" (https://arxiv.org/abs/cond-mat/9312044).

Also I would parametrize this type, i.e.:
```julia
struct BCS{T}
  data::T
end
```
so someone could dispatch on the type parameter if needed.